### PR TITLE
EOS-18771: Code changes to add x-forwarded headers while accessing csm_agent APIs

### PR DESCRIPTION
**Pull request by soniyamoholkar**
# UI

 Add forwarding info header to CSM web requests
## Problem Statement
<pre>
  <code>
    Story Ref (if any):EOS-18771
  </code>
</pre>
## Unit testing on RPM done
<pre>
  <code>
Yes
  </code>
</pre>
## Problem Description
<pre>
  <code>
Added client and proxy IP in headers as forwarded
  </code>
</pre>
## Solution/Screenshots
image
  </code>
</pre>Signed-off-by:

### CODE REVIEW

**Review by kupranay — APPROVED** [review]

File: web/src/services/api.ts [hunk 1]
```diff
@@ -343,6 +334,27 @@ export abstract class Api {
...
+                || req.connection?.remoteAddress;
+        if(client_ip && client_ip.includes("::ffff:")){
+            client_ip = client_ip.replace("::ffff:", "")
+        }
+        return client_ip;
+    }
+    private static setHeaders(req: Request, headers: any) {
+        const requestData = req && req.body ? JSON.stringify(req.body) : "";
+        var client_ip = Api.getClientIP(req);
+        let authorization = req.headers ? (req.headers.authorization ? req.headers.authorization : "") : "";
```
Comment (shrimetta) [inline web/src/services/api.ts:349]: No need to check req.headers every time. One time should be enough

File: web/src/services/api.ts [hunk 2]
```diff
@@ -343,6 +334,27 @@ export abstract class Api {
...
+                && req.headers['x-forwarded-for'].split(',').shift())
+                || req.connection?.remoteAddress;
+        if(client_ip && client_ip.includes("::ffff:")){
+            client_ip = client_ip.replace("::ffff:", "")
+        }
+        return client_ip;
+    }
+    private static setHeaders(req: Request, headers: any) {
+        const requestData = req && req.body ? JSON.stringify(req.body) : "";
+        var client_ip = Api.getClientIP(req);
```
Comment (shrimetta) [inline web/src/services/api.ts:348]: change var to const

File: web/src/services/api.ts [hunk 3]
```diff
@@ -343,6 +334,27 @@ export abstract class Api {
             });
         };
     }
+    private static getClientIP(req: Request) {
+        var client_ip = (typeof req.headers['x-forwarded-for'] === 'string'
+                && req.headers['x-forwarded-for'].split(',').shift())
+                || req.connection?.remoteAddress;
+        if(client_ip && client_ip.includes("::ffff:")){
```
Comment (shrimetta) [inline web/src/services/api.ts]: client_ip.includes("::ffff:") This check is not required

Comment (soniyamoholkar) [reply web/src/services/api.ts]: Done.

File: web/src/services/api.ts (hunk 2 above)
Comment (soniyamoholkar) [reply web/src/services/api.ts:348]: Done.

**Review by shrimetta — CHANGES_REQUESTED** [review]
Please address minor review comments

File: web/src/services/api.ts (hunk 1 above)
Comment (soniyamoholkar) [reply web/src/services/api.ts]: Done

**Review by shrimetta — APPROVED** [review]

**Pull request merged**